src/test/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/SPARQLStar2CypherTranslatorTest.java

```diff
@@ -884,6 +885,38 @@ public void translateBGPTest() {
                 translation);
     }
 
+    @Test
+    public void translateBGPwithJoinRewriteTest() {
```

Comment:
I think it would be helpful to also add a test that focuses only on the new `rewriteJoin` function; i.e., that test creates a Cypher query with equi-joins, passes that query directly to the `rewriteJoin` function, and then checks that the equi-joins have been rewritten correctly.